jaxrs-testing/src/test/java/com/facebook/airlift/jaxrs/testing/TestJaxrsThriftTestingHttpProcessor.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.airlift.jaxrs.testing;
+
+import com.facebook.airlift.http.client.HttpStatus;
+import com.facebook.airlift.http.client.Request;
+import com.facebook.airlift.http.client.testing.TestingHttpClient;
+import com.facebook.airlift.http.client.thrift.ThriftBodyGenerator;
+import com.facebook.airlift.http.client.thrift.ThriftProtocol;
+import com.facebook.airlift.http.client.thrift.ThriftResponseHandler;
+import com.facebook.airlift.jaxrs.thrift.ThriftMapper;
+import com.facebook.drift.codec.ThriftCodec;
+import com.facebook.drift.codec.ThriftCodecManager;
+import com.facebook.drift.codec.internal.compiler.CompilerThriftCodecFactory;
+import org.testng.annotations.Test;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+
+import java.net.URI;
+
+import static com.facebook.airlift.http.client.Request.Builder.prepareGet;
+import static com.facebook.airlift.http.client.Request.Builder.preparePost;
+import static com.facebook.airlift.http.client.thrift.ThriftRequestUtils.prepareThriftGet;
+import static com.facebook.airlift.http.client.thrift.ThriftRequestUtils.prepareThriftPost;
+import static com.google.common.net.HttpHeaders.ACCEPT;
+import static javax.ws.rs.core.HttpHeaders.CONTENT_TYPE;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
+
+@Test
+public class TestJaxrsThriftTestingHttpProcessor
+{
+    private static ThriftCodecManager codecManager = new ThriftCodecManager(new CompilerThriftCodecFactory(false));
```

Comment:
Avoid statically initializing these fields. Initialize them in `@BeforeClass` and close and nullify them in the `@AfterClass(alwaysRun=true)` to avoid memory leaks